azure/vm.sh

```diff
@@ -296,7 +310,437 @@ fn azure_vm_disk_attach(name, resgroup, diskID) {
+		echo "vm.backup.create: created snapshot id: "+$snapshotid
+	}
+
+	echo "vm.backup.create: starting VM"
+
+	azure_vm_start($vmname, $resgroup)
+
+	echo "vm.backup.create: backup finished with success, creating lock"
+
+	nodelete <= _azure_vm_backup_get_nodelete_lock($bkp_resgroup)
+
+	azure_lock_create($nodelete, "CanNotDelete", $bkp_resgroup)
+
+	readonly <= _azure_vm_backup_get_readonly_lock($bkp_resgroup)
+
+	azure_lock_create($readonly, "ReadOnly", $bkp_resgroup)
+
+	echo "vm.backup.create: created lock, finished with success"
+
+	return $bkp_resgroup
+}
+
+# azure_vm_backup_list returns the list of all backups
+# for the given vm name + prefix. They are the same parameters
+# you used to create the backup.
+#
+# The backup list is a list of resource groups names, where each
+# resource group is a backup.
+#
+# The list will be ordered, from the more recent to the oldest backup.
+fn azure_vm_backup_list(vmname, prefix) {
+	resgroups <= azure_group_get_names()
+
+	filtered = ""
+
+	for resgroup in $resgroups {
+		hasprefix <= echo $resgroup | -grep "^"+$prefix
+		hasvmname <= echo $hasprefix | -grep $vmname+"$"
+
+		if $status == "0" {
+			filtered = $filtered+$resgroup+"\n"
+		}
+	}
+
+	echo "vm.backup.list: got: [" + $filtered + "], ordering"
+	return _azure_vm_backup_order_list($filtered)
+}
+
+# azure_vm_backup_delete deletes a backup. This function
+# will also remove the locks that prevents backups deletion.
+fn azure_vm_backup_delete(backup_resgroup) {
+
+	dellock <= _azure_vm_backup_get_nodelete_lock($backup_resgroup)
+	readlock <= _azure_vm_backup_get_readonly_lock($backup_resgroup)
+
+	echo "backup delete: removing lock: " + $dellock
+	azure_lock_delete($dellock, $backup_resgroup)
+	echo "backup delete: removing lock: " + $readlock
+	azure_lock_delete($readlock, $backup_resgroup)
+
+	# WHY: Deleting locks is not synchronous, this is a VERY
+	# lame workaround while we don't fix the azure_lock_delete function.
+	lame_workaround_sleep = "5"
+	sleep $lame_workaround_sleep
+	echo "backup delete: locks removed, deleting resource group: " + $backup_resgroup
+	azure_group_delete($backup_resgroup)
+}
+
+# azure_vm_backup_recover will recover a previously generated
+# backup. The backup must have been generated using
+# azure_vm_backup_create, since a whole pattern on resource
+# naming will be required in the recovery process.
+#
+# The vminstance parameter is a instance of the vm object,
+# created with azure_vm_new and configured just like you
+# would do to create a new VM. The only main difference is
+# that no os disk should be configured, since the os disk and
+# datadisks will be obtained from the backup_resgroup.
+#
+# The backup_resgroup is the name of the resource group
+# where the disks are stored just as it is returned by
+# azure_vm_backup_create.
+#
+# It is an error to provide a vm instance that has not
+# a os type setted using azure_vm_set_ostype.
+fn azure_vm_backup_recover(instance, backup_resgroup) {
+
+	fn log(msg) {
```

Comment:
The oficial nash release that we have has a broken print, and on klb I'm only referencing tagged versions of nash :-)